catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -0,0 +1,171 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.geocoding.feature;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.WKTWriter;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.service.SecurityServiceException;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.security.common.Security;
+import org.codice.ddf.spatial.geocoding.FeatureExtractionException;
+import org.codice.ddf.spatial.geocoding.FeatureExtractor;
+import org.codice.ddf.spatial.geocoding.FeatureIndexer;
+import org.codice.ddf.spatial.geocoding.FeatureIndexingException;
+import org.codice.ddf.spatial.geocoding.GeoCodingConstants;
+import org.opengis.feature.simple.SimpleFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CatalogFeatureIndexer implements FeatureIndexer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CatalogFeatureIndexer.class);
+
+  private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTWriter::new);
+
+  private Security security = Security.getInstance();
+
+  private CatalogFramework catalogFramework;
+
+  private CatalogHelper catalogHelper;
+
+  private IndexCallback indexCallback;
+
+  private int extractionCount;
+
+  private boolean doCreate;
+
+  public void setSecurity(Security security) {
+    this.security = security;
+  }
+
+  public CatalogFeatureIndexer(CatalogFramework catalogFramework, CatalogHelper catalogHelper) {
+    this.catalogFramework = catalogFramework;
+    this.catalogHelper = catalogHelper;
+  }
+
+  @Override
+  public void updateIndex(
+      String resource, FeatureExtractor featureExtractor, boolean create, IndexCallback callback)
+      throws FeatureExtractionException, FeatureIndexingException {
+    Validate.notNull(featureExtractor, "featureExtractor can't be null");
+
+    doCreate = create;
+    indexCallback = callback;
```

Comment:
Is CatalogFeatureIndexer being called in a thread-safe environment?

Reply:
Fixed to be thread safe